devimint/src/federation.rs

```diff
@@ -265,148 +264,111 @@ impl Fedimintd {
     }
 }
 
-pub async fn run_dkg(root_task_group: &TaskGroup, servers: usize) -> anyhow::Result<()> {
-    async fn create_tls(id: usize, sender: Sender<String>) -> anyhow::Result<()> {
-        // set env vars
+pub async fn run_dkg(process_mgr: &ProcessManager, servers: usize) -> anyhow::Result<()> {
+    async fn create_tls(id: usize, env: &vars::Fedimintd) -> anyhow::Result<String> {
         let server_name = format!("Server {id}!");
-        let env_vars = fedimint_env(id)?;
-        let p2p_url = env_vars.get("FM_P2P_URL").context("FM_P2P_URL not found")?;
-        let api_url = env_vars.get("FM_API_URL").context("FM_API_URL not found")?;
-        let out_dir = env_vars
-            .get("FM_FEDIMINT_DATA_DIR")
-            .context("FM_FEDIMINT_DATA_DIR not found")?;
-        let cert_path = format!("{out_dir}/tls-cert");
-
-        // create out-dir
-        fs::create_dir(&out_dir).await?;
-
-        info!("creating TLS certs created for started {server_name} in {out_dir}");
+
         cmd!(
             "distributedgen",
             "create-cert",
-            "--p2p-url={p2p_url}",
-            "--api-url={api_url}",
-            "--out-dir={out_dir}",
+            "--p2p-url",
+            &env.FM_P2P_URL,
+            "--api-url",
+            &env.FM_API_URL,
+            "--out-dir",
+            utf8(&env.FM_DATA_DIR),
             "--name={server_name}",
         )
-        .envs(fedimint_env(id)?)
-        .run()
+        .envs(env.vars())
+        .run_log(format!("fedimintd-{id}"))
```

Comment:
This function name isn't clear to me. Maybe `run_and_log` or `run_with_logging`?